packages/varlet-ui/src/button/Button.vue

```diff
@@ -48,29 +48,44 @@ export default defineComponent({
   directives: { Ripple },
   props,
   setup(props) {
+    let loadingRef = props.autoLoading ? ref(false) : toRef(props, 'loading')
+
     const handleClick = (e: Event) => {
-      const { loading, disabled, onClick } = props
+      const { disabled, onClick, autoLoading } = props
 
-      if (loading || disabled) {
+      if (loadingRef.value || disabled) {
         return
       }
 
-      onClick?.(e)
+      const returnValue = onClick?.(e)
+      if (autoLoading && returnValue instanceof Promise) {
```

Comment:
The behavior of handling autoloading needs to be extracted into a function

Reply:
returnValue instanceof Promise should be extracted into a function called isPromise and put in utils/shared

Reply:
👌